### DISCUSSION TITLE
srt generation questions

### DISCUSSION THREAD

**Discussion by PacoH** [openai/whisper#1022] [Q&A]
I am amazed that `whisper` can generate `srt` files, and that it can accept video files, not only audio. The resulting `srt` file is much better than Youtube's CC generated subtitles with their annoying overlapping 'double lines', and more accurate as well. 

A couple of questions arose after generating subtitle files.

The output file includes the video file's extension (<filename.ext.srt>) instead of the expected <filename.srt>. This causes the player to not find any subtitles unless you remove the video extension. Is there a way to directly output the expected filename?

The lines of text, though highly accurate, are all lower case, with no punctuation, and are so long that the text occupies half the frame height in the case of a portrait mode frame. Is there a way to specify line length, and sentence case?

**Comment by ryanheise** [comment]
> The output file includes the video file's extension (<filename.ext.srt>) instead of the expected <filename.srt>. This causes the player to not find any subtitles unless you remove the video extension. Is there a way to directly output the expected filename?

There's an open pull request to address this: #1021 

Edit: #1021 has now been merged.

> The lines of text, though highly accurate, are all lower case, with no punctuation,

While I don't have it in front of me right now, this question has been asked a lot. If you search past discussions, you'll find many answers. (If you find one, feel free to add a link like the one above.)

> and are so long that the text occupies half the frame height in the case of a portrait mode frame.

Ditto.

**Reply by PacoH** [reply]
Thanks. I did a search of `subtitles` but found nothing that was simple or easy to implement.

**Reply by ryanheise** [reply]
Rather than search generally for `subtitles` you might have more success searching for each of the separate questions you're asking. E.g.

If you search for "split long lines" you'll find #314 
If you search for "no punctuation" you'll find #194

And since there is already some activity and progress in those discussions, you might consider continuing the respective discussions over there.
